### PR TITLE
Changeset operation documentation

### PR DESCRIPTION
**Pull request by austeja-bentley**
In this PR:
- Added documentation for Changeset operations in both @itwin/imodels-client-management and @itwin/imodels-client-authoring packages
- Changed `DownloadChangesetListParams` interface to allow `$skip` url parameter. This parameter is not very useful for `ChangesetOperations.downloadList` method but it unifies and simplifies the interface, now both `ChangesetOperations.downloadSingle` and `ChangesetOperations.downloadList` operations simply extend their corresponding entity query operation params with no exceptions.

### CODE REVIEW

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts [hunk 1]
```diff
@@ -21,52 +25,97 @@ export enum ContainingChanges {
...
 
+/** Synchronization information. */
 export interface SynchronizationInfo {
+  /** List of files that were processed by the synchronization. */
   changedFiles: string[];
 }
 
+/** DTO for links that belong to Changeset entity returned from iModels API. */
 export interface ChangesetLinks {
+  /** Link to upload the Changeset file. Points to file storage. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts:42]: Link where to upload the Changeset file. Link points to a remote storage.

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts [hunk 2]
```diff
@@ -21,52 +25,97 @@ export enum ContainingChanges {
...
   upload: Link;
+  /** Link to download the Changeset file. Points to file storage. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts:44]: Link from where to download the Changeset file. Link points to a remote storage.

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts [hunk 3]
```diff
@@ -21,52 +25,97 @@ export enum ContainingChanges {
...
+  /**
+   * Link to a Checkpoint that is created on a current or preceding Changeset. Points to a specific Checkpoint
+   * in iModels API.
+   * */
   currentOrPrecedingCheckpoint?: Link;
 }
 
+/** Minimal representation of a Changeset. */
 export interface MinimalChangeset {
+  /** Changeset id. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts:62]: Id of the Changeset

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts [hunk 4]
```diff
@@ -21,52 +25,97 @@ export enum ContainingChanges {
...
   id: string;
+  /** Changeset display name. */
   displayName: string;
+  /** Changeset description. */
   description: string;
+  /** Changeset index. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts:68]: Index of the Changeset

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts [hunk 5]
```diff
@@ -21,52 +25,97 @@ export enum ContainingChanges {
...
   index: number;
+  /** Changeset parent id. Equals to empty string if the Changeset is first in sequence. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts:70]: Id of the parent Changeset

File: clients/imodels-client-management/src/operations/changeset/ChangesetOperationParams.ts [hunk 6]
```diff
@@ -4,17 +4,34 @@
...
+  /** Filters Changesets which have an index less than or equal than specified in `lastIndex` property. For example,
+   * `lastIndex: 10` will return all Changesets that have an index less than or equal to 10. This option can be
+   * combined with {@link GetChangesetListUrlParams.afterIndex} option to query a specific Changeset range.
+   */
   lastIndex?: number;
 }
 
+/** Parameters for get Changeset list operation. */
 export interface GetChangesetListParams extends IModelScopedOperationParams {
+  /** Parameters that will be appended to the entity list request url that will narrow down or alter the results. */
```
Comment (robertas-kerpys) [inline clients/imodels-client-management/src/operations/changeset/ChangesetOperationParams.ts]: I would remove alter the results

**Review by robertas-kerpys — CHANGES_REQUESTED** [review]

File: clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts [hunk 7]
```diff
@@ -2,22 +2,39 @@
...
 import { TargetDirectoryParam } from "../../base";
 
+/** Properties that should be specified when creating a new Changeset. */
 export interface ChangesetPropertiesForCreate {
+  /** Changeset id. Changeset id must not be an empty or whitespace string. */
   id: string;
+  /** Changeset description. Changeset description must not exceed allowed 255 characters. */
   description?: string;
+  /**
+   * Parent Changeset id. Parent Changeset id must not be an empty or whitespace string except for when creating the
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts#L15

File: clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts [hunk 8]
```diff
@@ -2,22 +2,39 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-import { ContainingChanges, GetChangesetListUrlParams, GetSingleChangesetParams, IModelScopedOperationParams } from "@itwin/imodels-client-management";
+import { ContainingChanges, GetChangesetListParams, GetSingleChangesetParams, IModelScopedOperationParams } from "@itwin/imodels-client-management";
 import { TargetDirectoryParam } from "../../base";
 
+/** Properties that should be specified when creating a new Changeset. */
 export interface ChangesetPropertiesForCreate {
+  /** Changeset id. Changeset id must not be an empty or whitespace string. */
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts#L10

File: clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts [hunk 9]
```diff
@@ -2,22 +2,39 @@
...
   id: string;
+  /** Changeset description. Changeset description must not exceed allowed 255 characters. */
   description?: string;
+  /**
+   * Parent Changeset id. Parent Changeset id must not be an empty or whitespace string except for when creating the
+   * first Changeset - in such case it can be skipped or set to an empty string.
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-authoring/src/operations/changeset/ChangesetOperationParams.ts#L16

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts (hunk 1 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts#L42

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts (hunk 2 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts#L45

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts (hunk 3 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts#L62

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts (hunk 4 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts#L68

File: clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts (hunk 5 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/base/interfaces/apiEntities/ChangesetInterfaces.ts#L70

File: clients/imodels-client-management/src/operations/changeset/ChangesetOperationParams.ts (hunk 6 above)
Comment (austeja-bentley) [reply clients/imodels-client-management/src/operations/changeset/ChangesetOperationParams.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/1aac582dbc6b1ead914650a280cabe53a4f0487e/clients/imodels-client-management/src/operations/changeset/ChangesetOperationParams.ts#L34

**Review by robertas-kerpys — APPROVED** [review]

**Pull request merged**